theano/gpuarray/opt.py

```diff
@@ -1112,10 +1112,20 @@ def local_gpua_careduce(op, context_name, inputs, outputs):
         else:
             return False
         x, = inputs
+        idtype = x.dtype
```

Comment:
You should also need an opt for the classes Argmax and MaxAndArgmax. Otherwise, it won't be used on the GPU in the end.